pkg/virtctl/scp/native.go

```diff
@@ -0,0 +1,119 @@
+package scp
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/povsister/scp"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/ssh"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+func (o *SCP) nativeSCP(local templates.LocalSCPArgument, remote templates.RemoteSCPArgument, toRemote bool) error {
+	sshClient := ssh.NativeSSHConnection{
+		ClientConfig: o.clientConfig,
+		Options:      o.options,
+	}
+	client, err := sshClient.PrepareSSHClient(remote.Kind, remote.Namespace, remote.Name)
+	if err != nil {
+		return err
+	}
+
+	scpClient, err := scp.NewClientFromExistingSSH(client, &scp.ClientOption{})
+	if err != nil {
+		return err
+	}
+
+	if toRemote {
+		return o.copyToRemote(scpClient, local.Path, remote.Path)
+	}
+	return o.copyFromRemote(scpClient, local.Path, remote.Path)
+}
+
+func (o *SCP) copyToRemote(client *scp.Client, localPath, remotePath string) error {
+	isFile, isDir, exists, err := stat(localPath)
+	if err != nil {
+		return fmt.Errorf("failed reading path %q: %v", localPath, err)
+	}
+
+	if !exists {
+		return fmt.Errorf("local path %q does not exist, can't copy it", localPath)
+	}
+
+	if o.recursive {
+		if isFile {
+			return fmt.Errorf("local path %q is not a directory but '--recursive' was provided", localPath)
+		}
+
+		if err := client.CopyDirToRemote(localPath, remotePath, &scp.DirTransferOption{PreserveProp: o.preserve}); err != nil {
+			return err
```

Comment:
For simplify a bit the code you could return directly the result of the function `client.CopyDirToRemote`. If you return here you could also avoid the else and the last  ̇return nil ̇

Reply:
Obviously, I was sleeping here. :+1: